pkg/vault/instances.go

```diff
@@ -165,47 +176,69 @@ func configureMaster() string {
 		log.WithError(err).Fatal("failed to initialize master Vault client")
 	}
 
-	var clientToken string
-	switch authType := defaultGetenv("VAULT_AUTHTYPE", "approle"); strings.ToLower(authType) {
-	case APPROLE_AUTH:
-		roleID := mustGetenv("VAULT_ROLE_ID")
-		secretID := mustGetenv("VAULT_SECRET_ID")
-
-		secret, err := client.Logical().Write("auth/approle/login", map[string]interface{}{
-			"role_id":   roleID,
-			"secret_id": secretID,
-		})
+	masterAuthBundle := instanceCreds[masterVaultCFG.Address]
+	// indicates kube auth should be utilized
+	if len(masterAuthBundle.KubeRoleName) > 0 {
+		err := configureKubeAuthClient(client, masterAuthBundle, kubeSATokenPath)
 		if err != nil {
-			log.WithError(err).Fatal("[Vault Client] failed to login to master Vault with AppRole")
+			log.WithError(err).Fatal("[Vault Client] failed to configure master client using kube auth")
+		}
+	} else {
+		var clientToken string
+		switch authType := defaultGetenv("VAULT_AUTHTYPE", "approle"); strings.ToLower(authType) {
+		case APPROLE_AUTH:
+			roleID := mustGetenv("VAULT_ROLE_ID")
+			secretID := mustGetenv("VAULT_SECRET_ID")
+
+			secret, err := client.Logical().Write("auth/approle/login", map[string]interface{}{
+				"role_id":   roleID,
+				"secret_id": secretID,
+			})
+			if err != nil {
+				log.WithError(err).Fatal("[Vault Client] failed to login to master Vault with AppRole")
+			}
+			clientToken = secret.Auth.ClientToken
+		case TOKEN_AUTH:
+			clientToken = mustGetenv("VAULT_TOKEN")
+		default:
+			log.WithField("authType", authType).Fatal("[Vault Client] unsupported auth type")
 		}
-		clientToken = secret.Auth.ClientToken
-	case TOKEN_AUTH:
-		clientToken = mustGetenv("VAULT_TOKEN")
-	default:
-		log.WithField("authType", authType).Fatal("[Vault Client] unsupported auth type")
+		client.SetToken(clientToken)
 	}
 
-	client.SetToken(clientToken)
 	vaultClients[masterVaultCFG.Address] = client
 	return masterVaultCFG.Address
 }
 
+func configureKubeAuthClient(client *api.Client, bundle AuthBundle, kubeSATokenPath string) error {
+	kubeAuth, err := kubernetes.NewKubernetesAuth(
+		bundle.KubeRoleName,
+		kubernetes.WithServiceAccountTokenPath(kubeSATokenPath),
+	)
+	if err != nil {
+		return err
+	}
+	authInfo, err := client.Auth().Login(context.TODO(), kubeAuth)
```

Comment:
Why the usage of this context type here?

Reply:
Because I forgot to circle back and update it :sweat_smile: 

Good catch. Updated to use timeout